File: Frontend/TruckEase/TruckEaseApp/src/app/models/mappers/company.mapper.ts
import { CompanyInfoResponse } from "../responses/company-info.response";
import { CompanyInfoViewModel } from "../view-models/company-view-model";

export namespace CompanyMapper {
  export function toCompanyInfoViewModel(companyResponse: CompanyInfoResponse): CompanyInfoViewModel {

    return {
        id: companyResponse.id,
        companyName: companyResponse.companyName,
        registrationNumber: companyResponse.registrationNumber,
        companyType: companyResponse.companyType,
        description: companyResponse.description
  }

}
}

File: Backend/TruckEase/TruckEase/Contracts/Company/Mappers/CompanyMapper.cs
using TruckEase.Contracts.Company.Responses;
using TruckEase.Dtos;

namespace TruckEase.Contracts.Company.Mappers;

public static class CompanyMapper
{
    public static CompanyResponse ToCompanyInfoResponse(this CompanyInfoDto contactCompany)
    {
        return new CompanyResponse(
            contactCompany.Id,
            contactCompany.Name,
            contactCompany.RegistrationNumber,
            contactCompany.CompanyType,
            contactCompany.Description);
    }
}
